src/scenario.py

```diff
@@ -68,17 +66,17 @@ def run(self, strategy_name, options):
 
         event_steps = self.events.get_event_steps(self.start_time, self.n_intervals, self.interval)
 
-        socs = []
-        costs = []
-        prices = []
+        socs = {gcID: [] for gcID in self.constants.grid_connectors.keys()}
+        costs = {gcID: [] for gcID in self.constants.grid_connectors.keys()}
+        prices = {gcID: [] for gcID in self.constants.grid_connectors.keys()}
         results = []
-        extLoads = []
-        totalLoad = []
-        disconnect = []
-        feedInPower = []
+        extLoads = {gcID: [] for gcID in self.constants.grid_connectors.keys()}
+        totalLoad = {gcID: [] for gcID in self.constants.grid_connectors.keys()}
+        disconnect = {gcID: [] for gcID in self.constants.grid_connectors.keys()}
+        feedInPower = {gcID: [] for gcID in self.constants.grid_connectors.keys()}
         stepsPerHour = datetime.timedelta(hours=1) / self.interval
-        batteryLevels = {k: [] for k in self.constants.batteries.keys()}
-        connChargeByTS = []
+        batteryLevels = {gcID: {} for gcID in self.constants.grid_connectors.keys()}
```

Comment:
batteryLevels should still be saved by battery, not GC. When plotting, you may display the parent of the battery, but the values should be saved by battery. As it is, this line will not work, as you later try to append to nothing (you'd need to initialize an empty array for each battery of each GC)

Reply:
thats how it is actually implemented right now, see line 216